pkg/ioutils/readers.go

```diff
@@ -74,14 +93,61 @@ func NewBufReaderWithDrainbufAndBuffer(r io.Reader, drainBuffer []byte, buffer *
 }
 
 func (r *bufReader) drain() {
+	var (
+		duration       time.Duration
+		lastReset      time.Time
+		now            time.Time
+		reset          bool
+		bufLen         int64
+		dataSinceReset int64
+		maxBufLen      int64
+		reuseBufLen    int64
+		reuseCount     int64
+	)
+	reuseBufLen = int64(len(r.reuseBuf))
```

Comment:
Ok, I'll make the necessary changes. However, I wanted to make sure we don't wrap around if we've read more than 4 GB or if the buffer has grown above 4 GB since the last reset on 32 bit platforms.